web/src/main/java/com/microsoft/applicationinsights/web/internal/cookies/SessionCookie.java

```diff
@@ -191,8 +190,10 @@ private void parseCookie(Cookie cookie) throws Exception {
      */
     private Date parseDateWithBackwardCompatibility(String dateStr) throws ParseException {
         try {
-            return new Date(Long.parseLong(dateStr));
-        } catch (NumberFormatException e) {
+            Scanner s = new Scanner(dateStr);
+            s.useDelimiter("\\.");
```

Comment:
Is the date string always going to have separators as "." ? Can date string ever be in format like 01/01/2018 ? If that is the case then this logic will not work. We might need to consider all seperators.

Reply:
@dhaval24 can you please check what AppInsights Javascript library sends you? I mean this is contract between to microsoft appinsights codebases one for JS and one for Java. I didn't find any documentation about it (no surprise as this is internal concern). I am sure you can get in touch with relevant team? 

Note that this bug was introduced over a year ago when js team started sending ms in cookie. So this is not about date (date was implementation used even earlier and you have code for backwards compatibility below), now it's format is <long_for_datetime>.<miliseconds> from looking at JS code https://github.com/Microsoft/ApplicationInsights-JS/blob/cac7b363a7f59147adff510f698bc3f4943f2ab0/JavaScript/JavaScriptSDK/ajax/ajaxUtils.ts#L36 (and we see that being a problem in our test env). Is there some ms spec on how ai_session cookie is meant to be?